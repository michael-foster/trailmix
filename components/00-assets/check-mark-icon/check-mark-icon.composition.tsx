import React from 'react';
import { CheckMarkIcon } from './check-mark-icon';

export const BasicCheckMarkIcon = () => <CheckMarkIcon fillColor="black" />;
