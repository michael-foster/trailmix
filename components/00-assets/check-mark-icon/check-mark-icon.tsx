import React from 'react';
import { SVG } from '@outside-interactive/web-microfrontends.helpers.types';

export const CheckMarkIcon = ({
  classes = '',
  fillColor = '',
  strokeColor = '',
  strokeWidth = ''
}: SVG) => (
  <svg
    className={classes}
    viewBox="0 0 20 16"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1.94483 7.70578L0 9.69527L6.2069 16L20 1.98949L18.0552 0L6.2069 12.035L1.94483 7.70578Z"
      fill={fillColor}
      stroke={strokeColor}
      strokeWidth={strokeWidth}
    />
  </svg>
);
