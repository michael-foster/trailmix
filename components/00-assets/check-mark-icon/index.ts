export { CheckMarkIcon } from './check-mark-icon';
