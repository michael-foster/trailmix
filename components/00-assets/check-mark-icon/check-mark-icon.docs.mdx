---
description: 'A CheckMarkIcon component.'
labels: ['SVG', 'ASSET']
---

import { CheckMarkIcon } from './check-mark-icon';

## React Component for rendering a check mark SVG

### Component usage

```js
<CheckMarkIcon fillColor="#d2a940" />
```

### Using props to customize the icon

Modify the fill color to see it change live:

```js live
<CheckMarkIcon fillColor="#d2a940" />
```
