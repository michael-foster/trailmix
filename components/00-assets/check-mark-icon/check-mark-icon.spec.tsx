import React from 'react';
import { render } from '@testing-library/react';
import { BasicCheckMarkIcon } from './check-mark-icon.composition';

describe('CheckMarkIcon', () => {
  test('snapshot', () => {
    const { asFragment } = render(<BasicCheckMarkIcon />);

    expect(asFragment()).toMatchSnapshot();
  });
});
