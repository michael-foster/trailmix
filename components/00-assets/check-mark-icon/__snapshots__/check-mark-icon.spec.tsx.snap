// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckMarkIcon snapshot 1`] = `
<DocumentFragment>
  <svg
    class=""
    fill="none"
    viewBox="0 0 20 16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1.94483 7.70578L0 9.69527L6.2069 16L20 1.98949L18.0552 0L6.2069 12.035L1.94483 7.70578Z"
      fill="black"
      stroke=""
      stroke-width=""
    />
  </svg>
</DocumentFragment>
`;
